src/ResourceManager/Cdn/Commands.Cdn/Endpoint/NewAzureRmCdnEndpoint.cs

```diff
@@ -0,0 +1,132 @@
+    public class NewAzureRmCdnEndpoint : AzureCdnCmdletBase
+    {
+        [Parameter(Mandatory = true, HelpMessage = "Azure Cdn endpoint name.")]
+        [ValidateNotNullOrEmpty]
+        public string EndpointName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "Azure Cdn profile name.")]
+        [ValidateNotNullOrEmpty]
+        public string ProfileName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The resource group of the Azure Cdn Profile")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The location of the Cdn endpoint")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The origin host header of the Azure Cdn Endpoint")]
+        public string OriginHostHeader { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The origin path Azure Cdn Endpoint")]
+        public string OriginPath { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The list of mime types that need to be compressed by Cdn edge nodes")]
+        public string[] ContentTypesToCompress { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Is the compression enabled for the Cdn")]
+        public bool? IsCompressionEnabled { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Is the http traffic allowed for the Cdn")]
+        public bool? IsHttpAllowed { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Is the https traffic allowed for the Cdn")]
+        public bool? IsHttpsAllowed { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The way Cdn handles requests with query string")]
+        public PSQueryStringCachingBehavior? QueryStringCachingBehavior { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the origin used to identify the origin")]
+        public string OriginName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The host name of the origin")]
+        public string OriginHostName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The port http traffic used on the origin server")]
+        public int? HttpPort { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The port https traffic used on the origin server")]
+        public int? HttpsPort { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "The tags to associate with the Azure Cdn Endpoint")]
+        public Hashtable Tags { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                CdnManagementClient.Endpoints.GetWithHttpMessagesAsync(EndpointName, ProfileName, ResourceGroupName).Wait();
```

Comment:
Again, this seems like not the best pattern - you could use if-none-match to guarantee you are creating an endpoint that does not exist